src/OrleansTestingHost/TestCluster.cs

```diff
@@ -102,6 +102,16 @@ public TestCluster UseSiloBuilderFactory<TSiloBuilderFactory>() where TSiloBuild
             return this;
         }
 
+        private Func<ClientConfiguration, IClientBuilder> clientBuilderFactory;
+        /// <summary>
+        /// Set client builder factory, which would create a client builder to build the <see cref="TestCluster"/> client.
+        /// </summary>
+        public TestCluster UseClientBuilderFactory(Func<ClientConfiguration, IClientBuilder> clientBuilderFactory)
```

Comment:
I plan to revamp this and the silo builder approach very soon, but I think this is OK in the meantime, as it's similar to how currently the silo builder is configured.

Reply:
yeah feel free to revamp it., I just hook it up in the silobuilder fashion in current PR so I can test the new way to configure gateway